src/Sylius/Bundle/CartBundle/Provider/CartProvider.php

```diff
@@ -130,6 +138,15 @@ private function provideCart()
         }
 
         $cart = $this->cartFactory->createNew();
+
+        $event = new GenericEvent($cart);
+        $this->eventDispatcher->dispatch(SyliusCartEvents::CART_INITIALIZE, $event);
+
+        $cart = $event->getSubject();
+
+        $this->entityManager->persist($cart);
+        $this->entityManager->flush();
```

Comment:
I know cart identifier, but why don't create one so we don't have to flush database here?